src/tests/ftest/deployment/critical_integration.py

```diff
@@ -0,0 +1,108 @@
+                                  " 'echo hello'".format(host))
+            command_for_inter_node = ("clush --nostdin -S -w {}"
+                                      " 'echo hello'".format(','.join(self.hostlist_servers)))
+            try:
+                out = run_command(daos_server_cmd)
+                daos_server_version_list.append(out.stdout.split(b' ')[3])
+                if check_remote_root_access:
+                    run_command(remote_root_access)
+                IorTestBase._execute_command(self, command_for_inter_node, hosts=[host])
+            except (DaosTestError, CommandFailure) as error:
+                self.log.error("Error: %s", error)
+                failed_nodes.append(host)
+        if failed_nodes:
+            self.fail("Ssh check failed on following nodes.\n {}".format(failed_nodes))
+
+        for host in self.hostlist_clients:
+            dmg_version_cmd = ("ssh -oNumberOfPasswordPrompts=0 {}"
+                               " 'dmg version -i'".format(host))
+
+            try:
+                out = run_command(dmg_version_cmd)
+                dmg_version_list.append(out.stdout.split(b' ')[2])
+            except DaosTestError as error:
+                self.log.error("Error: %s", error)
+                failed_nodes.append(host)
+        if failed_nodes:
+            self.fail("SSH check for client nodes failed.\n {}".format(failed_nodes))
+
+        result_daos_server = (daos_server_version_list.count(daos_server_version_list[0])
+                              == len(daos_server_version_list))
+        result_dmg = dmg_version_list.count(dmg_version_list[0]) == len(dmg_version_list)
+        result_client_server = daos_server_version_list[0][1:] == dmg_version_list[0]
+
+        if (result_daos_server and result_dmg and result_client_server):
+            self.log.info("All servers have same daos version")
+            self.log.info("All clients have same daos version")
+            self.log.info("Servers and Clients have same daos version")
+        else:
+            self.log.info("Not all servers and clients have same daos version")
+            self.log.info("Clients: %s", dmg_version_list)
+            self.log.info("servers: %s", daos_server_version_list)
+            self.fail()
+
+    def test_ras(self):
+        """
+        Test Description: Verify RAS event on all server nodes from testrunner.
+                          Verify network scan and storage scan for server nodes.
+        :avocado: tags=all,deployment,full_regression
+        :avocado: tags=hw,large
+        :avocado: tags=criticalintegration,ras
+        """
+
+        dmg = self.get_dmg_command()
+        rank_list = self.server_managers[0].get_host_ranks(self.hostlist_servers)
+        print("rank_list: {}".format(rank_list))
+
+        for rank in rank_list:
+            dmg.system_stop(ranks=rank)
+            time.sleep(5)
+            dmg.system_start(ranks=rank)
+            time.sleep(5)
```

Comment:
How does this "Verify RAS event"?

Reply:
I believe starting/stopping ranks is considered one of the RAS events isn't it?

Reply:
I think I misunderstand. This code doesn't check for any RAS events in logs or output. So how do we know they are working? Does the simple act of stop+start implicitly mean RAS events are working?

Reply:
+1. How do we know appropriate RAS events are logged when sytem stop/start is done?

Reply:
The requirement is to trigger a RAS event on all Daos nodes on Aurora, not to verify in logs if the RAS event gets logged or not. So if we see a problem starting stopping any servers we know there is a problem.

Reply:
Also, if the server start/stop fails we should see it as a dmg command failure to know something is not right.

Reply:
Do you have a link/document for this requirement? I don't fully understand the intention here.

Reply:
The requirements can be mapped to the Test plan document for On-site testing available on teams [here](https://intel.sharepoint.com/:w:/r/sites/HPCStorageArchitectureandDevelopment/Shared%20Documents/On-Site%20Testing/On-Site%20Test%20Plan.docx?d=w89c919b20d844214bc22168034a8cc3f&csf=1&web=1&e=m6Fv83)
Intention here is to verify we can trigger a RAS event on every DAOS node.

Reply:
From the document:
> Servers involved in a given test are configured to forward RAS & telemetry data to HPE system management nodes.  When a new server is incorporated into a test cluster at least one RAS event is triggered to verify.
> ...
> All cluster members forward RAS and telemetry data to HPE management nodes.

So how do _**know**_ a RAS event is triggered and forwarded? We _**expect**_ start/stop to trigger a RAS event, but what if it doesn't and still returns successfully? E.g., nodes are somehow disconnected. TBH, I'm not very familiar with the logging and events in this area, so I'm not sure how we should check. Maybe something like this **_implicitly_** verifies nodes are reachable and therefore events were forwarded?
```
dmg system stop for each rank
dmg system query - verify Stopped
dmg system start for each rank
dmg system query - verify Joined
```
**_IF_** this is what we should do, there are existing utilities to help:
https://github.com/daos-stack/daos/blob/a4b34159fdaf482a03ae7c248fb2fda628850ae9/src/tests/ftest/util/server_utils.py#L742
https://github.com/daos-stack/daos/blob/a4b34159fdaf482a03ae7c248fb2fda628850ae9/src/tests/ftest/util/server_utils.py#L636

Reply:
Thanks for the suggestion. I have incorporated the suggestion in newer version. Instead of using 
``` def stop_ranks(self, ranks, daos_log, force=False):```
I have kept the original rank start/stop methods, as `def stop_ranks` also updates the expected state of the rank as part of the method and as I am restarting the same rank, I would have to add the extra step to again update the expected state of the same rank.
Also, regarding proposed
```def check_system_state(self, valid_states, max_checks=1):```
method to verify if ranks stopped/started, I have added a new method similar to the one proposed.
The problem with the proposed one was, that it was checking for the state of entire DAOS system and not a single rank in the system. Hence I had to add a new method which can check the state of single server rank.